File: src/main/java/com/lambda/javaorders/services/OrderServiceimpl.java
package com.lambda.javaorders.services;

import com.lambda.javaorders.models.Orders;
import com.lambda.javaorders.respositories.OrdersRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import javax.persistence.EntityNotFoundException;
import java.util.ArrayList;
import java.util.List;


@Service(value = "orderServices")
public class OrderServiceimpl implements OrderServices
{

    @Autowired
    private OrdersRepository orderrepo;

    @Override
   public List<Orders> findAllOrders(){
        List<Orders> rtnOrders = new ArrayList<>();
        orderrepo.findAll()
                .iterator()
                .forEachRemaining(rtnOrders::add);
        return rtnOrders;
    }
    @Override
   public Orders findOrderById(long id){
        return orderrepo.findById(id).orElseThrow(()-> new EntityNotFoundException("Order "+ id + " Not found"));
    }
}

File: src/main/kotlin/com/jakeesveld/orders/service/OrderServiceImpl.kt
package com.jakeesveld.orders.service

import com.jakeesveld.orders.Order
import com.jakeesveld.orders.OrdersRepository
import org.springframework.beans.factory.annotation.Autowired

class OrderServiceImpl: OrderService {

    @Autowired
    var repo: OrdersRepository? = null

    override fun findAll(): List<Order> {
        val orderList = mutableListOf<Order>()
        repo!!.findAll().iterator().forEachRemaining { order: Order? -> order?.let { orderList.add(it) } }
        return orderList
    }

    override fun findByCustomer(id: Long): List<Order> {
        val orderList = mutableListOf<Order>()
        repo!!.findAll().iterator().forEachRemaining{order: Order? -> order?.let { orderList.add(it) }}
        return orderList.filter { order -> order.custCode.custCode == id }
    }
}